### PR TITLE
[DDW-283] properly wait for node update

### PR DESCRIPTION
**Pull request by DominikGuzei**
This PR fixes the node update handling code to properly wait for the request.

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`npm run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`npm run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`npm run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`npm run flow:test`)
- [ ] There are no *lint* errors or warnings (`npm run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `npm run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`npm run storybook`)
- [ ] In case of npm dependency changes both `package-lock.json` and `yarn.lock` files are updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/renderer/app/stores/ada/NodeUpdateStore.js [hunk 1]
```diff
@@ -30,9 +30,9 @@ export default class NodeUpdateStore extends Store {
     setInterval(this.refreshNextUpdate, NODE_UPDATE_POLL_INTERVAL);
   }
 
-  @action refreshNextUpdate = () => {
+  @action refreshNextUpdate = async () => {
     if (this.stores.networkStatus.isSynced) {
-      this.nextUpdateRequest.execute();
+      await this.nextUpdateRequest.execute();
       if (this.nextUpdateRequest.result && !this.isUpdateAvailable &&
```
Comment (nikolaglumac) [inline source/renderer/app/stores/ada/NodeUpdateStore.js]: @DominikGuzei shouldn't we now wrap the code after await in the `runInAction` block?

Comment (DominikGuzei) [reply source/renderer/app/stores/ada/NodeUpdateStore.js]: Seems to be supported by mobx now: https://mobx.js.org/best/actions.html#async-await

Comment (nikolaglumac) [reply source/renderer/app/stores/ada/NodeUpdateStore.js]: Perpahs I was misunderstood. Don't you think the lines 37-39 should be turned into this:
```
runInAction(() => {
        this.isUpdateAvailable = true;
        this.isNotificationExpanded = true;
        this.updateVersion = this.nextUpdateRequest.result.version;
})
```
...this is the recommended way from the docs: https://mobx.js.org/best/actions.html#async-await
Or am I missing something? :)

Comment (DominikGuzei) [reply source/renderer/app/stores/ada/NodeUpdateStore.js]: Ah true … i missed the last sentence! 😉

**Comment by DominikGuzei** [comment]
@nikolaglumac i fixed the async mobx code now 👍 thanks for catching that mistake!

**Pull request merged**